packages/bpk-component-pagination/src/BpkPaginationList.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkPaginationPage from './BpkPaginationPage';
+import BpkPaginationBreak from './BpkPaginationBreak';
+import STYLES from './bpk-pagination-page-list.scss';
+
+const getClassName = cssModules(STYLES);
+
+const BpkPaginationList = (props) => {
+  const {
+    currentPage, pageCount, onPageChange, visibleRange,
+  } = props;
+
+  const children = [];
+  const shoulderRange = Math.ceil(visibleRange / 2);
+  const isFirstPageSelected = currentPage === 1;
+  const isLastPageSelected = currentPage === pageCount;
+  const displayRange = isFirstPageSelected || isLastPageSelected ? shoulderRange + 1 : shoulderRange;
+
+  for (let i = 1; i <= pageCount; i += 1) {
```

Comment:
This could be written as a map expression. `(new Array(pageCount)).map(_, i) => ...`. However this would work better if the whole component used zero-indexing which I also suggest we do

Reply:
This isn't 1-to-1 mapping, I thought about map expression but then I will have to filter out the null ones. Good point with 0-indexing though.

Reply:
Is that really problematic though? Just go

```javascript
const children = (new Array(pageCount)).map((_, i) => ...).filter(x => !!x)
```

Reply:
yeah, but even for mapping an empty array, you will still have to handle it like this https://stackoverflow.com/a/35086350, a bit of overkill, don't you think?

Reply:
Dunno I personally think it still reads a lot easier than the current imperative version with a for loop.